anyblok_pyramid/bloks/authorization/authorization.py

```diff
@@ -189,3 +190,478 @@ def check_validity(self):
             raise AuthorizationValidationException(
                 "Primary keys without model to apply in the authorization "
                 "(%s)" % self)
+
+    @classmethod
+    def ensure_authorizations_exists(
+        cls, role, model, authorizatoions
+    ):
+        """Ensure role's models authorizations are present in an
+        idempotent way
+        """
+        if not authorizatoions:
+            authorizatoions = {}
+        authz = (
+            cls.registry.Pyramid.Authorization.query()
+            .filter_by(
+                role=role,
+                model=model,
+            )
+            .one_or_none()
+        )
+        if not authz:
+            authz = cls.registry.Pyramid.Authorization.insert(
+                role=role,
+                model=model,
+            )
+        authz.update(
+            perms=authorizatoions.get("perms", {}),
+            manual=False,
+            **authorizatoions.get("extra_authz_params", {})
+        )
+        authz.flag_modified("perms")
+        return authz
+
+
+@Declarations.register(Pyramid)
+class Role:
+
+    @classmethod
+    def ensure_role_exists(
+        cls, name, config, label=None
+    ):
+        """Create or update Pyramid.Role with related model's authorization
+        in an idempotent way.
+
+        :param name: str, Role name
+        :param config: dict, per model authorizations configuration likes::
+
+            {
+                "Model.Test": {
+                    perms: {
+                        "create": {"matched": True},
+                        "read": {"matched": True},
+                        "update": {"matched": True},
+                        "delete": {"matched": True}
+                        "whatever you needs": {"matched": True}
+                    },
+                    extra_authz_params: {
+                        "order": 100,
+                    }
+                }
+            }
+
+        As a convention we suggest to provid a class method to return
+        this configuration that let other blok to easly improuve roles.
+
+
+        :param label: str, Role label used if role doesn't exits on insert.
+                      default: capitalized name.
+        :return: Created or updated role
+        """
+        Pyramid = cls.registry.Pyramid
+        role = Pyramid.Role.query().get(name)
+        if not role:
+            if not label:
+                label = name.capitalize()
+            role = Pyramid.Role.insert(name=name, label=label)
+
+        for model, authz in config.items():
+            Pyramid.Authorization.ensure_authorizations_exists(
+                role, model, authz
+            )
+        Pyramid.Authorization.query().filter_by(role=role).filter(
+            ~Pyramid.Authorization.model.in_(config.keys())
+        ).delete(synchronize_session="fetch")
+        return role
+
+
+@Declarations.register(Pyramid.Authorization)
+class Configuration:
```

Comment:
@jssuzanne I'm not sure `Configuration` is a good choice as far there are already some AnyBlok and Pyramid configuration ? Do you have any preferences ?

* Model.Pyramid.Authorization.**Configuration** (like it is right now I'm writing those line at commit cf304f38b25431d28233c4954f57613eea43f889)
* Model.Pyramid.Authorization.**Configurator** 
* Model.Pyramid.Authorization.**Setter**
* Model.Pyramid.Authorization.**Initializer**
* Model.Pyramid.Authorization.**Utils**
...

Reply:
I think a model is too much juste for some dict json

```
from anyblok_furetui import ACCESS_CRUD, ACCES_READ
from anyblok_furetui.common import ACCESS_CRUD, ACCES_READ
```

Reply:
I've made things shorter and exposed it as suggested